Wall index

andreorvalho opened the pull request:
This will provide all code necessary to show a wall and all its posts.

andreorvalho commented on config/routes.rb:
```diff
@@ -1,2 +1,5 @@
 Rails.application.routes.draw do
+  get 'wall/index'
```
This is not necessary resources walls already does this

andreorvalho commented on app/controllers/wall_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class WallController < ApplicationController
+  def show
+    post = Post.find(params[:id])
```
What is that you are trying to achieve with this line? What is your goal?

andreorvalho commented on app/models/wall.rb:
```diff
@@ -1,4 +1,4 @@
 class Wall < ActiveRecord::Base
   # Associations
-  has_many :posts, dependent: :destroy
+  has_many :posts, dependent: :destroy, order: 'dateg desc'
```
What is dateg?

andreorvalho commented on app/views/wall/index.html.erb:
```diff
@@ -0,0 +1,4 @@
+<h1>Wall#index</h1>
+<p>Find me in app/views/wall/index.html.erb</p>
+<% post.walls.each do |o| %>
+          <td><%= o.dateg %></td>
```
indentation is wrong.
You are also missing an end here. 
it should be:

```erb
<% post.walls.each do |o| %>
<% end %>
```

andreorvalho commented on db/schema.rb:
```diff
@@ -32,6 +32,14 @@
+    t.string   "email"
+    t.datetime "created_at"
+    t.datetime "updated_at"
+  end
```
why and how did this get here?

andreorvalho commented on spec/controllers/wall_controller_spec.rb:
```diff
@@ -0,0 +1,12 @@
+      get 'index'
+      expect(response).to be_success
+    end
+  end
```
this tests the index action that you have not yet defined on the wall controller!

andreorvalho commented:
> It has a syntax error in the "http://localhost:3000/wall/index"
<%= o.dateg %>
When I erased get 'wall/index' the result was an error

This sentence does not really make sense to me. But if you are trying to define the show action why are you accessing an url to the index?

jonybravo commented on db/migrate/20141012225644_create_posts.rb:
```diff
@@ -2,8 +2,8 @@ class CreatePosts < ActiveRecord::Migration
+      add_index :posts, [:wall_id, :created_at]
     end
   end
 end
```
turn back...

jonybravo commented on app/models/post.rb:
```diff
@@ -4,4 +4,7 @@ class Post < ActiveRecord::Base
+
+  # Chronologi
+  default_scope -> { order(created_at: :desc) }
 end
```
goingo to erase and create a post controller with a "def index order"

andreorvalho commented on app/models/wall.rb:
```diff
@@ -1,4 +1,4 @@
 class Wall < ActiveRecord::Base
   # Associations
-  has_many :posts, dependent: :destroy
+  has_many :posts, -> { order('created_at DESC') }, dependent: :destroy
```
so this is still providing a default scope!

The problem is the following:
- Everytime I do wall.posts, those posts will always come ordered by created at...
- I want that everytime I do wall.posts that it does not come with any order.

this here is what I told you it was a scope: http://guides.rubyonrails.org/active_record_querying.html#scopes

This makes it possible for us to have something we can re-use on the controller when we need, it works like a method that you apply to a collection of object like the one wall.posts returns. I will commit the changes with the correct code

andreorvalho commented:
@jonybravo I have squashed the commits and added corrections from my last comments.

I have also added the correct tests! please have a look! make all the questions when you feel confortable to go on I will merge the PR
